problem/src/main/java/org/zalando/problem/Status.java

```diff
@@ -263,11 +266,36 @@
     private final int code;
     private final String reason;
 
+    // Build a HashMap of all status keyed by their codes.
+    // Used in the `ofCode` factory method
+    private static Map<Integer, Status> allStatusByCode = new HashMap<>();
+    static {
+        for (Status s : Status.values()) {
+            allStatusByCode.put(s.getStatusCode(), s);
+        }
+    }
+
     Status(final int statusCode, final String reasonPhrase) {
         this.code = statusCode;
         this.reason = reasonPhrase;
     }
 
+    /**
+     * Creates a Status instance from the given code.
+     *
+     * @param code the HTTP code as a number
+     * @return the correct enum value for this status code.
+     * @throws IllegalArgumentException if the given code does not correspond to a known HTTP status.
+     */
+    public static Status ofCode(Integer code) {
+        if (allStatusByCode.containsKey(code)) {
+            return allStatusByCode.get(code);
```

Comment:
If you just use `get` and a null check then you only need to query the map once.